src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -161,6 +110,34 @@ public FileResource(FileService fileService, ResourceLoader resourceLoader, Lect
         return responseEntityForFilePath(FilePathService.getTempFilepath() + filename);
     }
 
+    /**
+     * POST /markdownFileUpload : Upload a new file for markdown.
+     *
+     * @param file The file to save
+     * @param keepFileName specifies if original file name should be kept
+     * @return The path of the file
+     * @throws URISyntaxException if response path can't be converted into URI
+     */
+    @PostMapping("/markdownFileUpload")
+    @PreAuthorize("hasAnyRole('ADMIN', 'INSTRUCTOR', 'TA', 'USER')")
+    public ResponseEntity<String> saveMarkdownFile(@RequestParam(value = "file") MultipartFile file, @RequestParam("keepFileName") Boolean keepFileName) throws URISyntaxException {
+        log.debug("REST request to upload file for markdown: {}", file.getOriginalFilename());
+        return handleSaveFile(file, keepFileName, true);
+    }
+
+    /**
+     * GET /files/markdown/:filename : Get the markdown file with the given filename
+     *
+     * @param filename The filename of the file to get
+     * @return The requested file, or 404 if the file doesn't exist
+     */
+    @GetMapping("/files/markdown/{filename:.+}")
+    @PreAuthorize("permitAll()")
```

Comment:
Does this mean that really every user can download those files? I don't think this would be a good solution.
Could we apply the same principle as for lecture downloads, so that only students within the same course (or students who have access to the exercise) can download markdown files?

Or would this be too complicated for now?

Reply:
Every user who knows the url of the file can access it. Adding authentication here would be quite hard to do since there is no good way (as far as I know) of adding the authentication to the code generated by markdown accessing the file.